{
    "version": "1.17", 
    "alias": "unittests", 
    "color": "#0052aa", 
    "mute": false, 
    "solo": false, 
    "meta_data": {
        "positions": {
            "/test": [
                0, 
                0
            ]
        }, 
        "collapse": {
            "/test": false
        }
    }, 
    "nodes": {
        "/": {
            "child_order": [
                "test"
            ], 
            "attrs": {
                "fail_fast": {
                    "type": "bool", 
                    "value": "True"
                }
            }, 
            "code": [
                "import unittest"
            ]
        }, 
        "/test": {
            "start_point": true, 
            "code": [
                "# Fixme: This is a hack to get the imports to work inside the test_plugin unittest", 
                "# I'm not really sure why this hack is needed, will fix later.", 
                "import nxt.plugins.file_fallbacks", 
                "", 
                "loader = unittest.TestLoader()", 
                "tests = loader.discover('${file::../}')", 
                "results = unittest.TestResult()", 
                "results.failfast = ${fail_fast}", 
                "tests.run(results)", 
                "", 
                "if results.failures or results.errors:", 
                "    for _, fail in results.failures:", 
                "        print(fail)", 
                "    for _, error in results.errors:", 
                "        print(error)", 
                "    raise Exception('{} unit tests failed! {} errors'.format(len(results.failures), ", 
                "                                                             len(results.errors)))", 
                "    ", 
                "print('All {} unit tests passed!'.format(results.testsRun))"
            ]
        }
    }
}